include/fuzzuf/algorithms/nautilus/grammartec/newtypes.hpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * fuzzuf
+ * Copyright (C) 2022 Ricerca Security
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+/**
+ * @file newtypes.hpp
+ * @brief Definitions of some simple types
+ * @author Ricerca Security <fuzzuf-dev@ricsec.co.jp>
+ */
+#ifndef FUZZUF_INCLUDE_ALGORITHMS_NAUTILUS_GRAMMARTEC_NEWTYPES_HPP
+#define FUZZUF_INCLUDE_ALGORITHMS_NAUTILUS_GRAMMARTEC_NEWTYPES_HPP
+
+#include <boost/range/irange.hpp>
+
+
+namespace fuzzuf::algorithm::nautilus::grammartec {
+
+class IDBase {
+public:
+  IDBase() : _id(0) {}
+  IDBase(size_t id) : _id(id) {}
+  size_t id() const { return _id; }
```

Comment:
the virtual descructor should be added

Reply:
Fixed in [2d7994e](https://github.com/fuzzuf/fuzzuf/pull/23/commits/2d7994e50b3bc9487e4bd280a42d5d1efc95410c)